File: sparkpython/pythondemo/demos/WordCount.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-

from pyspark import SparkContext

"""
单词出现频率统计
"""

sc = SparkContext("local", "WordCount")


def countString(lines):
    rdd = sc.parallelize(lines)
    result = rdd.flatMap(lambda x: x.split(" ")).countByValue()
    for key, value in result.items():
        print("{0}:\t{1}".format(key, value))


def countFile(fileIn, fileOut):
    rdd = sc.textFile(fileIn);
    words = rdd.flatMap(lambda x: x.split(" "))
    counts = words.map(lambda word: (word, 1)).reduceByKey(lambda x, y: x + y)
    counts.saveAsTextFile(fileOut)


if __name__ == "__main__":
    lines = ["pandas", "i like pandas", "i am neohope"]
    countString(lines)

    fileIn="../../../datain/TDOI.txt"
    fileOut = "../../../dataout/wordcount.py"
    countFile(fileIn, fileOut)

File: sparkscala/scalademo/src/main/scala/com/neohope/spark/demos/scala/WordCount.scala
package com.neohope.spark.demos.scala

import org.apache.spark._
import org.apache.spark.SparkContext._

/**
 * 单词出现频率统计
 */

object WordCount {
  
  val sc = new SparkContext("local", "WordCount")
  
  def countString(lines: List[String]){
    val input = sc.parallelize(lines)
    val words = input.flatMap(line => line.split(" "))
    val wc = words.countByValue()
    println(wc.mkString(","))
  }

  def countFile(fileIn: String, fileOut: String){
    val input = sc.textFile(fileIn)
    val words = input.flatMap(line => line.split(" "))
    val counts = words.map(word => (word, 1)).reduceByKey { case (x, y) => x + y }
    counts.saveAsTextFile(fileOut)
  }
  
  def main(args: Array[String]) {
    val lines=List("pandas", "i like pandas", "i am neohope")
    countString(lines)
    
    val fileIn="../../datain/TDOI.txt"
    val fileOut="../../dataout/wordcount.scala"
    countFile(fileIn, fileOut)
  }
}